unity-client/Assets/Scripts/MainScripts/DCL/Components/Video/Tests/VideoTests.cs

```diff
@@ -163,15 +171,140 @@ public IEnumerator VideoTextureVisibleStateIsSetCorrectlyWhenEntityIsRemoved()
             Assert.IsTrue(videoTexture.isVisible, "DCLVideoTexture should be visible");
 
             scene.RemoveEntity(ent1.entityId, true);
-            yield return null; //a frame to wait DCLVideoTexture update            
+            yield return null; //a frame to wait DCLVideoTexture update
 
             Assert.IsTrue(!videoTexture.isVisible, "DCLVideoTexture should not be visible ");
         }
 
+        [UnityTest]
+        public IEnumerator VolumeWhenVideoCreatedWithNoUserInScene()
+        {
+            // We disable SceneController monobehaviour to avoid its current scene id update
+            sceneController.enabled = false;
+
+            // Set current scene as a different one
+            CommonScriptableObjects.sceneID.Set("unexistent-scene");
+
+            DCLVideoTexture videoTexture = CreateDCLVideoTexture(scene, "it-wont-load-during-test");
+            yield return videoTexture.routine;
+
+            DecentralandEntity ent1 = TestHelpers.CreateSceneEntity(scene);
+            BasicMaterial ent1Mat = TestHelpers.SharedComponentCreate<BasicMaterial, BasicMaterial.Model>(scene, CLASS_ID.BASIC_MATERIAL, new BasicMaterial.Model() { texture = videoTexture.id });
+            TestHelpers.SharedComponentAttach(ent1Mat, ent1);
+            yield return ent1Mat.routine;
+
+            BoxShape ent1Shape = TestHelpers.SharedComponentCreate<BoxShape, BoxShape.Model>(scene, CLASS_ID.BOX_SHAPE, new BoxShape.Model());
+            yield return ent1Shape.routine;
+
+            TestHelpers.SharedComponentAttach(ent1Shape, ent1);
+            yield return null; //a frame to wait DCLVideoTexture update
+
+            // Check the volume
+            Assert.AreEqual(0f, videoTexture.texturePlayer.volume);
+        }
+
+        [UnityTest]
+        public IEnumerator VolumeWhenVideoCreatedWithUserInScene()
+        {
+            // We disable SceneController monobehaviour to avoid its current scene id update
+            sceneController.enabled = false;
+
+            // Set current scene with this scene's id
+            CommonScriptableObjects.sceneID.Set(scene.sceneData.id);
+
+            DCLVideoTexture videoTexture = CreateDCLVideoTexture(scene, "it-wont-load-during-test");
+            yield return videoTexture.routine;
+
+            DecentralandEntity ent1 = TestHelpers.CreateSceneEntity(scene);
+            BasicMaterial ent1Mat = TestHelpers.SharedComponentCreate<BasicMaterial, BasicMaterial.Model>(scene, CLASS_ID.BASIC_MATERIAL, new BasicMaterial.Model() { texture = videoTexture.id });
+            TestHelpers.SharedComponentAttach(ent1Mat, ent1);
+            yield return ent1Mat.routine;
+
+            BoxShape ent1Shape = TestHelpers.SharedComponentCreate<BoxShape, BoxShape.Model>(scene, CLASS_ID.BOX_SHAPE, new BoxShape.Model());
+            yield return ent1Shape.routine;
+
+            TestHelpers.SharedComponentAttach(ent1Shape, ent1);
+            yield return null; //a frame to wait DCLVideoTexture update
```

Comment:
Here its a bit safe (i.e. CI proof) to do `yield return new WaitForAllMessagesProcessed();`.

Reply:
totally, done.